conanfile.py

```diff
@@ -19,31 +25,81 @@ def requirements(self):
-        self.requires("zlib/[>=1.2.11 <2]")  #3b9e037ae1c615d045a06c67d88491ae # Also needed, so we can find zlib.h and co (+ pinning exactly is good)
+        self.requires(
+            "minizip/1.2.13"
+        )  # 0b5296887a2558500d0323c6c94c8d02 # This depends on zlib, and basically patches it
+        self.requires(
+            "zlib/[>=1.2.11 <2]"
+        )  # 3b9e037ae1c615d045a06c67d88491ae # Also needed, so we can find zlib.h and co (+ pinning exactly is good)
         self.requires("openssl/[>=3 <4]")
 
         if self.options.with_testing:
             self.requires("gtest/1.14.0")
         if self.options.with_benchmark:
             self.requires("benchmark/1.8.3")
 
+    # Let people provide their own CMake for now
+    # def build_requirements(self):
+    #     self.tool_requires("cmake/3.29.0")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
 
-    #def build_requirements(self):
-    #    self.tool_requires("cmake/3.22.6")
+        tc.cache_variables["BUILD_CLI"] = True
+        tc.cache_variables["BUILD_RUBY_BINDINGS"] = True
+        tc.cache_variables["BUILD_PYTHON_BINDINGS"] = True
+        tc.cache_variables["BUILD_PYTHON_PIP_PACKAGE"] = False
+
+        tc.cache_variables["BUILD_TESTING"] = bool(self.options.with_testing)
+        tc.cache_variables["BUILD_BENCHMARK"] = bool(self.options.with_benchmark)
+
+        tc.cache_variables["CPACK_BINARY_TGZ"] = True
+        tc.cache_variables["CPACK_BINARY_IFW"] = False
+        tc.cache_variables["CPACK_BINARY_DEB"] = False
+        if self.settings.build_type == "Release":
+            if is_apple_os(self) or self.settings.os == "Windows":
+                tc.cache_variables["CPACK_BINARY_IFW"] = True
+            else:
+                tc.cache_variables["CPACK_BINARY_DEB"] = True
+        tc.cache_variables["CPACK_BINARY_NSIS"] = False
+        tc.cache_variables["CPACK_BINARY_RPM"] = False
+        tc.cache_variables["CPACK_BINARY_STGZ"] = False
+        tc.cache_variables["CPACK_BINARY_TBZ2"] = False
+        tc.cache_variables["CPACK_BINARY_TXZ"] = False
+        tc.cache_variables["CPACK_BINARY_TZ"] = False
+
+        v = sys.version_info
+        if (v.major, v.minor) == (3, 8):
+            python_version = f"{v.major}.{v.minor}.{v.micro}"
+            self.output.info(
+                f"Setting PYTHON_VERSION and Python_ROOT_DIR from your current python: {python_version}, '{sys.base_prefix}'"
+            )
+            tc.cache_variables["PYTHON_VERSION"] = python_version
+            tc.cache_variables["Python_ROOT_DIR"] = Path(sys.base_prefix)
+        else:
+            self.output.warning(
+                "Your current python is not in the 3.8.x range, which is what we target.\n"
+                "You'll need to pass it properly when configuring CMake\n"
+                "via -DPYTHON_VERSION:STRING='3.8.xx' and -DPython_ROOT_DIR:PATH='/path/to/python3.8/'"
```

Comment:
Try to infer the Python info. If you use python 3.8, all is good, that'd be passed to cmake.

I'm using base_prefix and not exec_prefix or prefix, because I want a Virtualenv python to resolve to the underlying python (system python or pyenv python)